app/Main.hs

```diff
@@ -1,27 +1,32 @@
 import System.Environment (getArgs)
 import System.Exit (exitFailure)
 
+import Data.Semigroup ((<>))
 import Data.Text (unpack)
+import qualified Options.Applicative as Opt
+import Options.Applicative ((<**>))
 
 import Krank
 import Krank.Formatter
 
-main :: IO ()
-main = do
-  filePath <- parseArgs
-  case filePath of Just f -> putStrLn . unpack . showViolations $ processFile f
-                   Nothing -> printHelp >> exitFailure
+data Sample = Sample {
+  codeFilePath :: FilePath
+}
+
+fileToParse :: Opt.Parser FilePath
+fileToParse = Opt.argument Opt.str (Opt.metavar "FILE")
 
-parseArgs :: IO (Maybe FilePath)
-parseArgs = do
-  args <- getArgs
-  case args of [a] -> pure $ Just a
-               _   -> pure Nothing
+sample :: Opt.Parser Sample
+sample = Sample
+  <$> fileToParse
```

Comment:
Question of style, I'd prefer if `fileToParse` was inlined here.

Reply:
You mean writing the statement on a single line?

Reply:
Yes.

Reply:
No. ;)

Reply:
I meant:

```
sample = Sample 
   <$> the_code_of_file_to_parse_directly_here.
```